src/OaiPmh/Harvester.php

```diff
@@ -168,7 +163,7 @@ public function launch()
         // Load last state, if applicable (used to recover from server failure).
         if ($state = $this->stateManager->loadState()) {
             $this->write("Found saved state; attempting to resume.\n");
-            [$resumeSet, $resumeToken, $this->startDate] = $state;
+            [$resumeSet, $resumeToken, $this->startDate, $this->harvestEndDate] = $state;
```

Comment:
For backward compatibility, I wonder if we should account for the possibility of $this->harvestEndDate being undefined here (in case a user upgrades and then resumes an interrupted harvest from a prior version). This is probably a pretty rare case, and maybe the thing to do is just throw an exception saying the harvest needs to be restarted.. but it might be worth accounting for the possibility in some way.

Reply:
While I also suspect that's a pretty rare case, it certainly could not hurt. Perhaps checking to see that we have a state array with 4 values will be most useful, as that would catch any strange situations where an incomplete state was saved for any reason.

Reply:
Works for me!

Reply:
https://github.com/vufind-org/vufindharvest/pull/8/commits/cb0d8cc360010573e1a8e1d47c92c3469bfe14ab addresses this piece.